internal/cmd/request.go

```diff
@@ -0,0 +1,107 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/api/server"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type CommonOptions struct {
+	Verbose bool `help:"Use verbose output to show full request/response without specific formatting." default:"false"`
+}
+
+func MustHandleCMDRequestAndExit(endpoint string, request any, responseType any) error {
+	return MustHandleCMDRequestAndExitFull(endpoint, request, responseType, CommonOptions{}, nil)
+}
+
+func MustHandleCMDRequestAndExitFull(endpoint string, request any, responseType any, options CommonOptions, customPrintFunc any) error {
+	response, err := SendCMDRequest(endpoint, request)
+	if err != nil {
+		return fmt.Errorf("Failed to send the CMD request: '%w'.", err)
+	}
+
+	if !response.Success {
+		fmt.Println(response.Message)
```

Comment:
No, you can't do this.
This message needs to be accessible to someone.
When printed it's not on stderr (like the logs) and not accessible via an error,
so it's just gone to anyone who is not capturing stdout (which we already talked about how bad that is to do for most other code in most cases).